pkg/operator2/apiservices/sync_take_over_apiservices_oauth_apiserver.go

```diff
@@ -0,0 +1,50 @@
+package apiservices
+
+import (
+	"context"
+	"fmt"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	operatorconfigclient "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+	operatorclientinformers "github.com/openshift/client-go/operator/informers/externalversions"
+)
+
+// NewManageAPIServicesController sets ManagingOAuthAPIServer flag to true.
+// This will make OpenShift APIServer to step down and traffic will be routed to the OAuth APIServer
+//
+// see https://github.com/openshift/enhancements/blob/master/enhancements/authentication/separate-oauth-resources.md
+func NewManageAPIServicesController(
+	name string,
+	authOperatorClient operatorconfigclient.AuthenticationsGetter,
+	authOperatorInformers operatorclientinformers.SharedInformerFactory,
+	eventRecorder events.Recorder) factory.Controller {
+
+	controllerFactory := factory.New()
+	authOperatorLister := authOperatorInformers.Operator().V1().Authentications().Lister()
+
+	controllerFactory.WithSync(func(ctx context.Context, controllerContext factory.SyncContext) error {
+		operator, err := authOperatorLister.Get("cluster")
+		if err != nil {
+			return err
+		}
+		if !v1helpers.IsOperatorConditionTrue(operator.Status.Conditions, "Available") {
+			message := "authentication operator is not Available"
+			controllerContext.Recorder().Warning("PrereqNotReady", message)
+			return fmt.Errorf(message)
+		}
+
+		if !operator.Status.ManagingOAuthAPIServer {
+			operatorCopy := operator.DeepCopy()
+			operatorCopy.Status.ManagingOAuthAPIServer = true
```

Comment:
@sttts if we decide to land it in `4.5` we would have to flip it to `false` in `4.4` (downgrade scenario). So that `OAS` can claim all `groups`

Reply:
unless we need to add it `OAS-O` for `4.4` because it runs before `CAO`?

Reply:
so, yeah `OAS-O` is run before `CAO`, so I think we need a way to tell `CAO` to step down otherwise they will be fighting with yeah other, am I right?

Reply:
We need a way to decide the fight for OAS to win as long as it is the latest version (don't forget rollback when there is an old OAS again).

Reply:
yeah, I am asking how it will behave on downgrades. As I understand it, we will have an old `OAS-O` and new `CAO`. Because `OAS-O` runs on a different runlevel that `CAO`

I think we need a way to claim back the API as soon as we will successfully roll back `OAS-O` and we need to tell the new `CAO` to step back?

Otherwise, they will be fighting until `CAO` is rolled back.